nion/swift/test/DocumentController_test.py

```diff
@@ -1177,6 +1181,51 @@ def test_get_3_data_sources_with_two_selected_in_data_panel(self):
             self.assertEqual(2, len(document_controller._get_n_data_sources(2)))
             self.assertEqual(3, len(document_controller._get_n_data_sources(3)))
 
+    def test_filenames_in_exports(self):
+        # The bmp writer is a dumb writer, and used to avoid complex setup code
+        FILE_NAMES = ["test.bmp", "/././////.test.bmp", "$$%/%%_test.bmp", "\\aaaaa//test.bmp"]
```

Comment:
There are still quite a few failing cases that aren't covered. Some examples on macOS are:

`FILE_NAMES = ["a\N{WARNING SIGN}b", "\N{WARNING SIGN}", ".", "abc\n\rdef", "a\tb", "µscope"]`

Also, valid files should not start with a `.` (period or dash).